controllers/activemqartemis_controller_test.go

```diff
@@ -1645,7 +1644,291 @@ var _ = Describe("artemis controller", func() {
 		})
 	})
 
-	Context("With delopyed controller", func() {
+	Context("With a deployed controller", func() {
+		//TODO: Remove the 4x duplication and add all acceptor settings
+		It("Testing acceptor keyStoreType being set", func() {
+			By("By creating a new custom resource instance")
+			ctx := context.Background()
+			cr := generateArtemisSpec(namespace)
+
+			cr.Spec.DeploymentPlan = brokerv1beta1.DeploymentPlanType{
+				Size: 1,
+			}
+			keyStoreType := "PKCS12"
+			cr.Spec.Acceptors = []brokerv1beta1.AcceptorType{
+				{
+					Name:         "new-acceptor",
+					Port:         61666,
+					SSLEnabled:   true,
+					KeyStoreType: keyStoreType,
+				},
+			}
+			Expect(k8sClient.Create(ctx, &cr)).Should(Succeed())
+
+			Eventually(func() bool {
+				key := types.NamespacedName{Name: cr.Name, Namespace: namespace}
+				err := k8sClient.Get(ctx, key, &cr)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+
+			key := types.NamespacedName{Name: cr.Name, Namespace: namespace}
+			Eventually(func() bool {
+				_, ok := namespacedNameToFSM[key]
+				return ok
+			}, timeout, interval).Should(BeTrue())
+
+			fsm := namespacedNameToFSM[key]
```

Comment:
the tests need to restrict them selves to the kube api, they can be remote from the controller. They can use the namers but their own instance, they should not depend on the FSM. There are other examples to peek at in this file.

Reply:
eg: https://github.com/artemiscloud/activemq-artemis-operator/blob/bcad426a7dfba540a7fbfadf2d1e85ce13743656/controllers/activemqartemis_controller_test.go#L1994

Reply:
some of the mods to the related tests are complete in: https://github.com/artemiscloud/activemq-artemis-operator/pull/184/files#diff-d78c1465cfcf5a1ea5e8e62663d05d50bc3d12175de63981c05441f8b31636dfR1703

Reply:
I disagree that we should be keeping ourselves to the kube API in tests. It is also important to test the other structural code within the operator even if it is indirect supporting code for the main interaction with kubernetes and the brokers. What was your thinking behind a restriction for test code to only cover the kube API usage?

Reply:
we'll try improve that in the future.

Reply:
for the kube api, that is only for the envTests, those start a kube api server to behave like a sub set of real kube. They sould work whether kube is local or remote, hense just use the kube api.
Other tests (more unit tets) can for sure use internal apis... but the point of the envTests is to have a bit of kube present.

Reply:
for sure it may make sense to have more test modules and split out envTests.

Reply:
here is a good point of reference for envTests: https://github.com/kubernetes-sigs/controller-runtime/blob/master/FAQ.md#q-how-should-i-write-tests--any-suggestions-for-getting-started